pygfunction/properties.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import, division, print_function
+
+from CoolProp.CoolProp import PropsSI
+import warnings
+
+
+class Fluid:
+    """
+        An object for handling the fluid properties
+
+        Parameters
+        ----------
+        m_flow: float
+            The fluid mass flow rate per borehole (kg/s)
+        T: float, optional
+            The temperature of the fluid (Kelvin)
+        P: float, optional
+            The pressure of the fluid (Pascal)
+        mixer: str, optional
+            The mixer for this application should be one of
+
+                - 'MEG' - Ethylene glycol mixed with water
+                - 'MPG' - Propylene glycol mixed with water
+                - 'MMA' - Methanol mixed with water
+                - 'MEA' - Ethanol mixed with water
+        percent: int, optional
+            The percentage of the mixing fluid added to the water.
+            Lower bound = 0
+            Upper bound is dependent on the mixture
+
+        Examples
+        ----------
+        >>> import pygfunction as gt
+        >>> m_flow = 0.25  # Total fluid mass flow rate per borehole (kg/s)
+        >>> T = 20 + 273  # Temp at 20 C for now to match GLHEPRO
+        >>> gage_P = 20  # PsiG
+        >>> atm_P = 14.69595
+        >>> P = (gage_P + atm_P) * 6894.75728  # Pressure in Pa
+
+        >>> # pure water
+        >>> fluid = gt.properties.Fluid(m_flow, T=T, P=P)
+        >>> print(fluid.__repr__())
+
+        >>> # 20 % propylene glycol mixed with water
+        >>> mix = 'MPG'
+        >>> percent = 20
+        >>> fluid = gt.properties.Fluid(m_flow, T=T, P=P, mixer=mix, percent=percent)
+
+        >>> # 60% ethylene glycol mixed with water
+        >>> mix = 'MEG'
+        >>> percent = 60
+        >>> fluid = gt.properties.Fluid(m_flow, T=T, P=P, mixer=mix, percent=percent)
+        >>> print(fluid.__repr__())
+
+        >>> # 5% methanol mixed with water water
+        >>> mix = 'MMA'
+        >>> percent = 5
+        >>> fluid = gt.properties.Fluid(m_flow, T=T, P=P, mixer=mix, percent=percent)
+        >>> print(fluid.__repr__())
+
+        >>> # ethanol / water
+        >>> mix = 'MEA'
+        >>> percent = 10
+        >>> fluid = gt.properties.Fluid(m_flow, T=T, P=P, mixer=mix, percent=percent)
+        >>> print(fluid.__repr__())
```

Comment:
Is there a difference between `print(fluid.__repr__())` and `print(fluid)`?

Reply:
If the `__repr__()` function did not exist, then `print(fluid)` has the following output:
```
<pygfunction.properties.Fluid object at 0x000001D5B26FFAC8>
```
However, with the `__repr__(self)` method uncommented, `print(fluid)` has this output:
```
<class 'pygfunction.properties.Fluid'>
P = 239220.0
Pr = 7.070547362075505
T = 293.0
cp = 4186.2937048371
fluid_mix = INCOMP::MEG-0%
k = 0.5988756481542833
mu = 0.0010114862770798584
nu = 1.0141456799573551e-06
rho = 997.3776914598615
rhoCp = 4175.315951103378
```
The next commit will just have `print(fluid)` because with the `__repr__` method defined, they result in the same.